### PR TITLE
Packager

### PR DESCRIPTION
**Pull request by nickschuch**
Our first major component! :)

This component is the open source version of our packager.

This will "build" and "push" multiple containers in the one build.

Example output looks like this:

Note: The frontend and backend prefixes are actually different colors.

carbon

To take it for a spin, see `packager/example`

### CODE REVIEW

**Comment by nickschuch** [comment]
Im going to make this take an `io.Writer` so that we can run a test suite against it eg.

* Build the images via a test
* Check the output

File: packager/config.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package packager
+
+import "github.com/pkg/errors"
+
+// Config holds our build configuration.
+type Config struct {
+	Project     string      `json:"project" yaml:"project"`
+	Push        bool        `json:"push" yaml:"push"`
```
Comment (nicksantamaria) [inline packager/config.go]: Is it possible to vertically align the yaml WKT definitions?

File: packager/config_test.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConfigValidate(t *testing.T) {
+	config := Config{}
+
+	err := config.Validate()
+	assert.Equal(t, err.Error(), "project not found")
```
Comment (nicksantamaria) [inline packager/config_test.go]: Can we add some positive test cases in here too? i.e. including a value returns `nil`

File: packager/packager_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
...1b[0m Step 1/1 : FROM busybox\n\x1b[38;5;1mfrontend\x1b[0m latest: Pulling from library/busybox\n\x1b[38;5;1mfrontend\x1b[0m 67a0688b88df: Pulling fs layer\n\x1b[38;5;1mfrontend\x1b[0m 67a0688b88df: Verifying Checksum\n\x1b[38;5;1mfrontend\x1b[0m 67a0688b88df: Download complete\n\x1b[38;5;1mfrontend\x1b[0m 67a0688b88df: Pull complete\n\x1b[38;5;1mfrontend\x1b[0m Digest: sha256:7a7714572b84fe1fafbc6b4f8374925249da8642b723d941a5ea214dac19b164\n\x1b[38;5;1mfrontend\x1b[0m Status: Downloaded newer image for busybox:latest\n\x1b[38;5;1mfrontend\x1b[0m  ---> f9b6f7f7b9d3\n\x1b[38;5;1mfrontend\x1b[0m Successfully built f9b6f7f7b9d3\n\x1b[38;5;1mfrontend\x1b[0m Successfully tagged example/project:0.0.1-frontend\n\x1b[38;5;2mbackend\x1b[0m Step 1/1 : FROM busybox\n\x1b[38;5;2mbackend\x1b[0m latest: Pulling from library/busybox\n\x1b[38;5;2mbackend\x1b[0m Digest: sha256:7a7714572b84fe1fafbc6b4f8374925249da8642b723d941a5ea214dac19b164\n\x1b[38;5;2mbackend\x1b[0m Status: Image is up to date for busybox:latest\n\x1b[38;5;2mbackend\x1b[0m  ---> f9b6f7f7b9d3\n\x1b[38;5;2mbackend\x1b[0m Successfully built f9b6f7f7b9d3\n\x1b[38;5;2mbackend\x1b[0m Successfully tagged example/project:0.0.1-backend\n"
```
Comment (nicksantamaria) [inline packager/packager_test.go]: Are these image hashes going to change over time?

Comment (nickschuch) [reply packager/packager_test.go]: Dammit! Your right!!!

File: packager/config_test.go (hunk 2 above)
Comment (nickschuch) [reply packager/config_test.go]: The check for nil on line 44 covers this?

File: packager/packager_test.go (hunk 3 above)
Comment (nickschuch) [reply packager/packager_test.go]: Im going to target specific strings within this output with `strings.Contains()`

File: Makefile [hunk 4]
```diff
@@ -8,4 +8,8 @@ PROJECT=github.com/previousnext/skpr
 lint:
 	golint -set_exit_status `go list ./... | grep -v /vendor/`
 
-.PHONY: lint
+# Run tests with coverage reporting
+test:
+	go test -cover ./packager/...
```
[the remark this answers is not in the document]
Comment (nicksantamaria) [reply Makefile]: This looks sorted!

File: packager/config_test.go (hunk 2 above)
Comment (nicksantamaria) [reply packager/config_test.go]: True!

**Comment by nicksantamaria** [comment]
LGTM!

**Pull request merged**